### ISSUE TITLE
Size controls inaccessible

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3444]
The size control buttons underneath the main edit textarea for making the textarea bigger, smaller and making the text wrap are inaccessible to keyboard only users, screen reader users and people with visual impairments.

![](https://user-images.githubusercontent.com/108893/111910436-3fada880-8a59-11eb-9122-63424c120bae.png)

## Keyboard

The buttons cannot be operated with the keyboard alone, you cannot tab to them.
That means neither people with motor impairments who cannot use the mouse or use voice control nor most screen reader users will be able to use them.

That fails [2.1.1 Keyboard](https://www.w3.org/TR/WCAG21/#keyboard). Although that is a complete barrier, the functionality itself is not that important, which is why this has only a medium priority.

## No accessible name

The buttons are missing an alternative text, which fails [1.1.1 Non-text Content](https://www.w3.org/TR/WCAG21/#non-text-content).
And they therefore don't have an accessible name, which fails [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

That means screen reader users who cannot see won't be able to perceive the buttons at all. And it makes it harder for voice control users to activate them.

## Colour contrast

The buttons have a colour contrast ratio of 1.8:1 (`#c0c0c0` against `#ffffff`) which is insufficient to meet the minimum requirement of 3:1.
That fails [1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG21/#non-text-contrast).

That means people with visual impairments will have trouble seeing the buttons.

## How to fix it

There are multiple ways to fix these issues. I would suggest to...

1. put the images into buttons; they don't need to visually look any different, although allowing them to look like buttons would make them more consistent and would give you the focus styles automatically
2. add alternative text to the images describing what they do (and add that to the language files); you could also make the text visible to everyone and leave the `alt` empty
3. make the colour of the images darker
4. ideally also add a border to the icon outlines (which could be white, so invisible in the default settings) so that the buttons don't disappear when users change colours
